stravalib/model.py

```diff
@@ -59,26 +57,49 @@
 from stravalib.strava_model import (
     Split,
     SportType,
+    SummaryClub,
     SummaryPRSegmentEffort,
     SummarySegmentEffort,
     TimedZoneRange,
 )
 
+if TYPE_CHECKING:
+    from stravalib.client import BatchedResultsIterator
+
 LOGGER = logging.getLogger(__name__)
 
 
-def lazy_property(fn):
+# TODO: how do you properly type this with more precision than Any?
+def lazy_property(fn: Callable[Any], Any) -> property:
```

Comment:
This line is actually the cause of all the failing test I think. It should read. After applying the modification from my other comment, all test pass.
```suggestion
def lazy_property(fn: Callable[[Any], Any]) -> property:
```

Reply:
oh ok! thanks! oh yes i see that's just wrong syntax.